package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -407,117 +443,177 @@ def guess_hydrogens(self,
-            hydrogen bonds.
+            String containing the :attr:`resname` and :attr:`name` of all atoms 
+            that potentially capable of forming hydrogen bonds.
 
         Notes
         -----
-        This function makes use of and atomic charges to identify which atoms could be considered acceptor atoms in the
-        hydrogen bond analysis. If an atom has an atomic charge less than :attr:`max_charge` then it is considered
-        capable of participating in hydrogen bonds.
+        Acceptor selections may be achieved with either a resname, atom 
+        name combination, or when those are absent, atom types.
 
-        If :attr:`acceptors_sel` is `None`, this function is called to guess the selection.
+        This function makes use of and atomic charges to identify which atoms 
+        could be considered acceptor atoms in the hydrogen bond analysis. If 
+        an atom has an atomic charge less than :attr:`max_charge` then it is 
+        considered capable of participating in hydrogen bonds.
 
-        Alternatively, this function may be used to quickly generate a :class:`str` of potential acceptor atoms involved
-        in hydrogen bonding. This :class:`str` may then be modified before being used to set the attribute
-        :attr:`acceptors_sel`.
+        If :attr:`acceptors_sel` is `None`, this function is called to guess 
+        the selection.
+
+        Alternatively, this function may be used to quickly generate a 
+        :class:`str` of potential acceptor atoms involved in hydrogen bonding. 
+        This :class:`str` may then be modified before being used to set the 
+        attribute :attr:`acceptors_sel`.
+
+
+        .. versionchanged:: 2.4.0
+            Added ability to use atom types
 
         """
 
         ag = self.u.select_atoms(select)
         acceptors_ag = ag[ag.charges < max_charge]
-        acceptors_list = np.unique(
-            [
-                '(resname {} and name {})'.format(r, p) for r, p in zip(acceptors_ag.resnames, acceptors_ag.names)
-            ]
-        )
 
-        return " or ".join(acceptors_list)
+        return self._group_categories(acceptors_ag)
+
+    @staticmethod
+    def _group_categories(group):
+        """ Find categories according to universe constraints
+        
+        Parameters
+        ----------
+        group : AtomGroup
+            AtomGroups corresponding to either hydrogen bond acceptors, 
+            donors, or hydrogen atoms that meet their respective charge
+            and mass constraints. 
+
+        Returns
+        -------
+        select : str
+            String for each hydrogen bond acceptor/donor/hydrogen atom category.
+
+
+        .. versionadded:: 2.4.0
+
+        """
+
+        if hasattr(group, "resnames") and hasattr(group, "names"):
+            group_list = np.unique([
+                '(resname {} and name {})'.format(r, 
+                    p) for r, p in zip(group.resnames, group.names)
+            ])
+        else:
```

Comment:
I still think we should have the option to _choose_ whether or not to default to types here, however for the sake of getting this done I'm going to let it go as-is here, and raise a separate issue to deal with later.